src/event.rs

```diff
@@ -35,24 +35,16 @@ use sodiumoxide::crypto::sign;
 #[derive(Clone, Eq, PartialEq)]
 pub enum Event {
     Request {
-        request        : Request,
+        request        : ExternalRequest,
         our_authority  : Authority,
         from_authority : Authority,
-        response_token : Vec<u8>,
+        response_token : SignedToken,
```

Comment:
The Response has the "original request token", not the request;

the upper layer also needs to be told the MessageId - to hand it back down in the Routing::Put, ::Get, etc